compiler/qsc_formatter/src/formatter/tests.rs

```diff
@@ -0,0 +1,307 @@
+                    span: Span {
+                        lo: 143,
+                        hi: 152,
+                    },
+                    new_text: "\n    ",
+                },
+                Edit {
+                    span: Span {
+                        lo: 176,
+                        hi: 189,
+                    },
+                    new_text: "\n        ",
+                },
+                Edit {
+                    span: Span {
+                        lo: 199,
+                        hi: 212,
+                    },
+                    new_text: "\n        ",
+                },
+                Edit {
+                    span: Span {
+                        lo: 222,
+                        hi: 236,
+                    },
+                    new_text: "\n\n        ",
+                },
+                Edit {
+                    span: Span {
+                        lo: 246,
+                        hi: 259,
+                    },
+                    new_text: "\n        ",
+                },
+                Edit {
+                    span: Span {
+                        lo: 268,
+                        hi: 277,
+                    },
+                    new_text: "\n    ",
+                },
+                Edit {
+                    span: Span {
+                        lo: 278,
+                        hi: 291,
+                    },
+                    new_text: "\n",
+                },
+            ]
+        "#]],
+    );
+}
+
+#[test]
+fn correct_empty_delimiters() {
+    check(
+        indoc! {r#"
+        operation Foo() : Unit {
+        }
+        operation Bar() : Unit {
+            operation Baz() : Unit {   }
+            let x = {
+
+            };
+            let y : Int[] = [ ];
+            let z = (
+
+             );
+        }
+        "#},
+        &expect![[r#"
+            [
+                Edit {
+                    span: Span {
+                        lo: 24,
+                        hi: 25,
+                    },
+                    new_text: "",
+                },
+                Edit {
+                    span: Span {
+                        lo: 80,
+                        hi: 83,
+                    },
+                    new_text: "",
+                },
+                Edit {
+                    span: Span {
+                        lo: 98,
+                        hi: 104,
+                    },
+                    new_text: "",
+                },
+                Edit {
+                    span: Span {
+                        lo: 128,
+                        hi: 129,
+                    },
+                    new_text: "",
+                },
+                Edit {
+                    span: Span {
+                        lo: 145,
+                        hi: 152,
+                    },
+                    new_text: "",
+                },
+            ]
+        "#]],
+    );
+}
+
+#[test]
+fn test_sample() {
```

Comment:
What is being asserted? What is being tested? What is the expected output?

Reply:
I've renamed this to `sample_has_no_formatting_changes` for clarity.